### DISCUSSION TITLE
the CPU & RAM usage of Github Copilot in Remote SSH environment in VSCode

### DISCUSSION THREAD

**Discussion by PwuJdu** [community/community#27237] [Copilot Conversations]
As we know that the copilot will take off many CPU and RAM resource. If I connnect a server through vscode remote ssh and install copilot in the server environment , which machine's CPU & RAM usage will be taken ? the local one( control ) or the remote one (be controled )?

**Comment by GAtwork** [comment]
It would appear that the workload is done by the remote machine. However, some instances of copilot idle or have very low memory and cpu usage on the local machine. It would most definitely (in my experience) cause memory usage increase on the remote machine and one should provision extra resources.

**Comment by hasteagag** [comment]
I just tried to spin up copilot on a shared hosting server on godaddy, and even after upgrading to 'ultimate' shared hosting, I hit the limit on both ram and process count almost immediately and got disconnected.  The actual cpu usage is very low, but the other demands appear to be around 1 gb in ram?  process count hits 140 almost immediately.  I never thought about it when running on WSL so I might give that a go while watching and report back.  It is just something to be aware of if your hosting platform or remote server has resource limits.
